recipes/configs/llama3_2_vision/11B_lora_td.yaml

```diff
@@ -0,0 +1,131 @@
+# Config for multi-device LoRA finetuning in lora_finetune_distributed_td.py
+# using a Llama3.2 11B Vision Instruct model
+#
+# This config assumes that you've run the following command before launching:
+#   tune download meta-llama/Llama-3.2-11B-Vision-Instruct --output-dir /tmp/Llama-3.2-11B-Vision-Instruct --ignore-patterns "original/consolidated*"
+#
+# To launch on 2 devices, run the following command from root:
+#   tune run --nproc_per_node 2 lora_finetune_distributed_td --config llama3_2_vision/11B_lora_td
+#
+# You can add specific overrides through the command line. For example
+# to override the checkpointer directory while launching training:
+#   tune run --nproc_per_node 2 lora_finetune_distributed_td --config llama3_2_vision/11B_lora_td checkpointer.checkpoint_dir=<YOUR_CHECKPOINT_DIR>
+#
+# This config works best when the model is being fine-tuned on 2+ GPUs.
+# For single device LoRA finetuning please use 11B_lora_single_device.yaml
+# or 11B_qlora_single_device.yaml
+
+# Model arguments
+model:
+  _component_: torchtune.models.llama3_2_vision.lora_llama3_2_vision_11b
+  decoder_trainable: "frozen"
+  encoder_trainable: "lora"
+  fusion_trainable: "lora"
+  lora_attn_modules: ['q_proj', 'v_proj']
+  apply_lora_to_mlp: False
+  apply_lora_to_output: False
+  lora_rank: 8
+  lora_alpha: 16
+  lora_dropout: 0.0
+  image_size: 560 # Make sure this matches the image_size in tokenizer
+
+# Transform
+tokenizer:
+  _component_: torchtune.models.llama3_2_vision.llama3_2_vision_transform
+  path: /tmp/Llama-3.2-11B-Vision-Instruct/original/tokenizer.model
+  image_size: 560
+  max_seq_len: 8192
+
+# Checkpointer
+checkpointer:
+  _component_: torchtune.training.FullModelHFCheckpointer
+  checkpoint_dir: /tmp/Llama-3.2-11B-Vision-Instruct/
+  checkpoint_files:
+    filename_format: model-{}-of-{}.safetensors
+    max_filename: "00005"
+  recipe_checkpoint: null
+  output_dir: /tmp/Llama-3.2-11B-Vision-Instruct/
+  model_type: LLAMA3_VISION
+resume_from_checkpoint: False
+save_adapter_weights_only: False # PeFT formatting not available yet. This will save it in torchtune format only.
+
+# TorchData Dataset setup
+dataloader:
```

Comment:
Usually libraries have dataset and dataloader as two stand alone modules. Torch also assumes that. I think that we should follow the same pattern. .e.g

```
dataloader:
	shuffle: True
	  collate_fn: torchtune.data.padded_collate_tiled_images_and_mask
	  parallel_method: thread
	  num_workers: 0  # Per dataset
	  pin_memory: true
	  packed: False # Set to true for great speed ups

datasets:
      ocrvqa:
      weight: 1.
      dataset:
        _component_: torchtune.datasets.multimodal.the_cauldron_dataset_torchdata
        subset: ocrvqa
    dvqa:
      weight: 1.
      dataset:
        _component_: torchtune.datasets.multimodal.the_cauldron_dataset_torchdata
        subset: dvqa
```

Reply:
Thanks for the comments! The rationale here is that settings like shuffle and workers might eventually be configured individually in the datasets, however I can change this if you think it's the right abstraction for torchtune. Happy to discuss more if you'd like